torchvision/transforms/functional_tensor.py

```diff
@@ -659,6 +659,9 @@ def _apply_grid_transform(img: Tensor, grid: Tensor, mode: str) -> Tensor:
         need_cast = True
         img = img.to(torch.float32)
 
+    if grid.device.type != img.device.type:
+        grid = grid.to(img)
```

Comment:
In PyTorch in general we don't perform implicit CPU / CUDA conversions, but we instead raise errors to the user, in order to avoid silent performance penalties.

I think we should instead fix the locations where the grid is created, so that they take the original Tensor device as input